File: frontend/classes/Office.ts
interface OfficeData {
  id: number
  senator: number
  name: string
  start_step: number
  end_step: number | null
}

class Office {
  id: number
  senator: number
  name: string
  start_step: number
  end_step: number | null

  constructor(data: OfficeData) {
    this.id = data.id
    this.name = data.name
    this.senator = data.senator
    this.start_step = data.start_step
    this.end_step = data.end_step
  }
}

export default Office

File: backend/rorapp/models/office.py
from django.db import models
from rorapp.models.senator import Senator


# Model for representing offices
class Office(models.Model):
    name = models.CharField(max_length=21)
    senator = models.ForeignKey(Senator, on_delete=models.CASCADE)
    start_step = models.IntegerField()
    end_step = models.IntegerField(blank=True, null=True)
    
    # String representation of the office, used in admin site
    def __str__(self):
        return f'{self.name} {self.senator}'
